internal/controller/finrestore_controller_test.go

```diff
@@ -486,6 +545,150 @@ var _ = Describe("FinRestore Controller Reconcile Test", Ordered, func() {
+			restoreJobKey := client.ObjectKey{Namespace: namespace, Name: restoreJobName(finrestore)}
+			Eventually(func(g Gomega, ctx SpecContext) {
+				var restoreJob batchv1.Job
+				g.Expect(k8sClient.Get(ctx, restoreJobKey, &restoreJob)).To(Succeed())
+				makeJobSucceeded(&restoreJob)
+				g.Expect(k8sClient.Status().Update(ctx, &restoreJob)).To(Succeed())
+			}, "5s", "1s").WithContext(ctx).Should(Succeed())
+
+			By("reconciling to process job completion")
+			_, err = reconciler.Reconcile(ctx, ctrl.Request{NamespacedName: client.ObjectKeyFromObject(finrestore)})
+			Expect(err).ShouldNot(HaveOccurred())
+
+			By("checking that FinRestore becomes ReadyToUse")
+			Eventually(func(g Gomega) {
+				var updated finv1.FinRestore
+				err := k8sClient.Get(ctx, client.ObjectKeyFromObject(finrestore), &updated)
+				g.Expect(err).ShouldNot(HaveOccurred())
+				g.Expect(updated.IsReady()).Should(BeTrue())
+			}, "5s", "1s").Should(Succeed())
+		})
+
+		It("should not become ReadyToUse when allowChecksumMismatched is false", func(ctx SpecContext) {
+			// Description:
+			//   Ensure that when a FinBackup has ChecksumMismatched=True and FinRestore has allowChecksumMismatched=false,
+			//   FinRestore does not become ReadyToUse.
+			//
+			// Arrange:
+			//   - Create a FinBackup with ChecksumMismatched=True.
+			//   - Create a FinRestore with allowChecksumMismatched=false.
+			//
+			// Act:
+			//   Run reconciliation.
+			//
+			// Assert:
+			//   - FinRestore does not become ReadyToUse.
+
+			// Arrange
+			By("creating FinBackup with ChecksumMismatched=True")
+			finbackup := CreateFinBackupStoredAndVerified(ctx, k8sClient, namespace, utils.GetUniqueName("test-fin-backup"), pvc, 1, utils.GetUniqueName("test-node"))
+			meta.SetStatusCondition(&finbackup.Status.Conditions, metav1.Condition{
+				Type:    finv1.BackupConditionChecksumMismatched,
+				Status:  metav1.ConditionTrue,
+				Reason:  "ChecksumMismatch",
+				Message: "Checksum corruption detected",
+			})
+			Expect(k8sClient.Status().Update(ctx, finbackup)).Should(Succeed())
+
+			By("creating FinRestore with allowChecksumMismatched=false")
+			finrestore := NewFinRestore(
+				namespace,
+				utils.GetUniqueName("test-restore"),
+				finbackup.Name,
+				utils.GetUniqueName("restore-pvc"),
+				namespace,
+			)
+			finrestore.Spec.AllowChecksumMismatched = false
+			Expect(k8sClient.Create(ctx, finrestore)).Should(Succeed())
+
+			// Act
+			By("reconciling the FinRestore")
+			_, err := reconciler.Reconcile(ctx, ctrl.Request{NamespacedName: client.ObjectKeyFromObject(finrestore)})
+			Expect(err).ShouldNot(HaveOccurred())
+
+			// Assert
+			By("checking that FinRestore does not become ReadyToUse")
+			Consistently(func(g Gomega) {
+				var updated finv1.FinRestore
+				err := k8sClient.Get(ctx, client.ObjectKeyFromObject(finrestore), &updated)
+				g.Expect(err).ShouldNot(HaveOccurred())
+				g.Expect(updated.IsReady()).Should(BeFalse())
+			}, "3s", "1s").Should(Succeed())
```

Comment:
This check seems indirect because it won't become ready without a restoration PV. Shouldn't we check the PVC won't be created?

Reply:
> but it might already exist when the mismatch occurs

I'm not quite sure about this. In this test `ChecksumMismatched` of the FinBackup becomes true (L.587) before FinRestore is created (L.598). So there's no chance that the restoration PV exists when the checksum mismatch has occurred.